### ISSUE TITLE
kwokctl create cluster --runtime=binary doesn't work on darwin arm64 (M1)

### ISSUE THREAD

**Issue by veggiemonk** [kubernetes-sigs/kwok#591]
### How to use it?

- [X] kwok
- [ ] kwokctl --runtime=docker (default runtime)
- [X] kwokctl --runtime=binary
- [ ] kwokctl --runtime=nerdctl
- [ ] kwokctl --runtime=kind

### What happened?

creating a cluster on mac M1 without docker involved by using the runtime flag

```shell
% kwokctl create cluster --runtime=binary
Cluster is creating                                                                                                     cluster=kwok
Download                                        cluster=kwok uri="https://dl.k8s.io/release/v1.27.1/bin/darwin/arm64/kube-apiserver"
ERROR Failed to setup config     cluster=kwok err="https://dl.k8s.io/release/v1.27.1/bin/darwin/arm64/kube-apiserver: 404 Not Found"
Cluster is cleaned up                                                                                                   cluster=kwok
ERROR Execute exit                            err="https://dl.k8s.io/release/v1.27.1/bin/darwin/arm64/kube-apiserver: 404 Not Found"
```

### What did you expect to happen?

a cluster is created

### How can we reproduce it (as minimally and precisely as possible)?

On a M1 mac:
kwokctl create cluster --runtime=binary

### Anything else we need to know?

possible solution: "build binaries locally"

```bash
wget https://dl.k8s.io/v1.27.1/kubernetes-src.tar.gz
tar xzf kubernetes-src.tar.gz
make WHAT=cmd/kube-apiserver
make WHAT=cmd/kube-controller-manager
make WHAT=cmd/kube-scheduler
cp _output/local/bin/darwin/arm64/* ~/.kwok/cache/XXX
```

### Kwok version

<details>

```console
$ kwok --version
kwok version v0.2.0 go1.20.3 (darwin/arm64)

$ kwokctl --version
kwokctl version v0.2.0 go1.20.3 (darwin/arm64)
```

</details>

### OS version

<details>

```console
# On Darwin:
$ uname -a
Darwin XXX 22.3.0 Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000 arm64
```

</details>

**Comment by wzshiming** [comment]
#269

This is a known issue, and I'm struggling with the best way to implement it

**Comment by veggiemonk** [comment]
Let me know what you are struggling with,maybe I can help 😄

**Comment by veggiemonk** [comment]
I'll be looking into number 2. I don't know how much management that entails.
Thank you for clarifying.

**Comment by wzshiming** [comment]
> Another options is to point people to documentation when they run the command with binaries not available for their platform.
> Like a gist: https://gist.github.com/veggiemonk/711c5010a3217a2f25f076164822dcfa
> Once the binaries are built, it is not too far fetched to let the user manage them.

Sound good to me, we can do this for Windows

On Darwin, I prefer to add the subcommand `kwokctl build binary`, which builds the binary and configures the `kwok` configuration.

**Comment by KunWuLuan** [comment]
On amd64 I meet the same problem

**Comment by wzshiming** [comment]
@KunWuLuan Yep, I will get the best way to implement it.

**Comment by wzshiming** [comment]
Fixed by #627

https://kwok.sigs.k8s.io/docs/user/kwokctl-platform-specific-binaries/

/close

**Comment by wzshiming** [comment]
At present, pre-built UNOFFICIAL binary has been provided for Darwin and Windows, you can refer to this for configuration to use.

https://kwok.sigs.k8s.io/docs/user/kwokctl-platform-specific-binaries/#use-unofficial-binaries
